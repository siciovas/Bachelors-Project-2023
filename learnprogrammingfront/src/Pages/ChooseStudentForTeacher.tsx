import React, { useCallback } from "react";
import { useState, useEffect } from "react";
import {
  Table,
  Thead,
  Tbody,
  Tr,
  Th,
  Td,
  Text,
  Box,
  Heading,
  Spinner,
  Flex,
  Input,
  Avatar,
  InputGroup,
  InputLeftElement,
  Icon,
  Button,
  useDisclosure,
  Modal,
  ModalBody,
  ModalCloseButton,
  ModalContent,
  ModalFooter,
  ModalHeader,
  ModalOverlay,
} from "@chakra-ui/react";
import { SearchIcon } from "@chakra-ui/icons";
import { UserTypes } from "../Types/UserTypes";
import eventBus from "../Helpers/EventBus";
import { Unauthorized } from "../Constants/Auth";
import toast from "react-hot-toast";

const ChooseStudentForTeacher = () => {
  const token = localStorage.getItem("accessToken");
  const [users, setUsers] = useState<UserTypes[]>([]);
  const [isLoading, setIsLoading] = useState(true);
  const [searchTerm, setSearchTerm] = useState("");
  const { isOpen, onOpen, onClose } = useDisclosure();
  const [assignId, setAssignId] = useState<string>();

  const openModal = (id: string) => {
    setAssignId(id);
    onOpen();
  };
  const filteredUsers = users.filter(
    (user) =>
      user.name.toLowerCase().includes(searchTerm.toLowerCase()) ||
      user.surname.toLowerCase().includes(searchTerm.toLowerCase()) ||
      user.email.toLowerCase().includes(searchTerm.toLowerCase())
  );

  const getUsers = useCallback(async () => {
    const response = await fetch(`https://localhost:7266/api/getUsers`, {
      headers: {
        "Content-Type": "application/json",
        Authorization: `Bearer ${token}`,
      },
      method: "GET",
    });
    if (response.status === 401) {
      eventBus.dispatch("logOut", Unauthorized);
    } else if (response.status === 200) {
      const user = await response.json();
      setUsers(user);
      setIsLoading(false);
    } else {
      toast.error("Netikėta klaida!");
    }
  }, []);

  useEffect(() => {
    getUsers();
  }, [isLoading]);

  const assignStudent = async (
    e: React.MouseEvent<HTMLButtonElement>
  ): Promise<void> => {
    e.preventDefault();
    const response = await fetch("https://localhost:7266/api/postStudent", {
      headers: {
        "Content-Type": "application/json",
        Authorization: `Bearer ${token}`,
      },
      method: "POST",
      body: JSON.stringify({
        studentId: assignId,
      }),
    });
    if (response.status === 401) {
      eventBus.dispatch("logOut", Unauthorized);
    }
    if (response.status === 200) {
      setIsLoading(true);
      toast.success("Studentas priskirtas!");
      onClose();
    } else {
      toast.error("Nepavyko!");
    }
  };

  if (isLoading) {
    return (
      <Flex justifyContent="center" top="50%" left="50%" position="fixed">
        <Spinner size="xl" />
      </Flex>
    );
  }
  return (
    <>
      <Box p={4}>
        <Flex mb={8} justify="center">
          <Heading size="lg">Studentų sąrašas</Heading>
        </Flex>
        <Flex justify={"center"} width={"25%"} margin={"auto"}>
          <InputGroup mb={4}>
            <InputLeftElement
              pointerEvents="none"
              children={<Icon as={SearchIcon} color="white" />}
            />
            <Input
              type="text"
              placeholder="Ieškokite pagal vardą ir pavardę"
              value={searchTerm}
              onChange={(event) => setSearchTerm(event.target.value)}
            />
          </InputGroup>
        </Flex>
        <Box overflowX="auto" maxWidth="100%">
          <Table variant="striped">
            <Thead>
              <Tr>
                <Th>Nuotrauka</Th>
                <Th>Asmuo</Th>
                <Th>El.Paštas</Th>
                <Th>Miestas</Th>
                <Th>Mokykla</Th>
                <Th>Statusas</Th>
              </Tr>
            </Thead>
            <Tbody>
              {filteredUsers.map((user, index) => (
                <Tr key={index}>
                  <Td>
                    <Avatar src={"data:image/jpeg;base64," + user.avatar} />
                  </Td>
                  <Td>
                    <Text>
                      {user.name} {user.surname}
                    </Text>
                  </Td>
                  <Td>
                    <Text>{user.email}</Text>
                  </Td>
                  <Td>
                    <Text>{user.city}</Text>
                  </Td>
                  <Td>
                    <Text>{user.school}</Text>
                  </Td>
                  <Td>
                    {user.isAssigned === false ? (
                      <Button
                        colorScheme={"green"}
                        borderRadius={"50px 50px 50px 50px"}
                        onClick={() => openModal(user.id)}
                        _hover={{
                          bg: "green",
                        }}
                      >
                        Priskirti studentą
                      </Button>
                    ) : (
                      <Text>Užimtas</Text>
                    )}
                  </Td>
                </Tr>
              ))}
            </Tbody>
          </Table>
        </Box>
      </Box>
      <>
        <Modal isOpen={isOpen} onClose={onClose}>
          <ModalOverlay />
          <ModalContent>
            <ModalHeader>Perspėjimas!</ModalHeader>
            <ModalCloseButton />
            <ModalBody>
              <Text>Ar tikrai norite priskirti studentą į savo komandą?</Text>
            </ModalBody>

            <ModalFooter>
              <Button
                background="green.500"
                color={"white"}
                mr={3}
                borderRadius={"50px 50px 50px 50px"}
                onClick={(e) => assignStudent(e)}
                _hover={{
                  bg: "green",
                }}
              >
                Priskirti
              </Button>
            </ModalFooter>
          </ModalContent>
        </Modal>
      </>
    </>
  );
};

export { ChooseStudentForTeacher };
