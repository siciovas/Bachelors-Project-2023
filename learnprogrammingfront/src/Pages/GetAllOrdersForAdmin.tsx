import React from "react";
  
  const GetAllOrdersForAdmin = () => {
  
    return (
        <h1>adminas visus uzsakymus matys cia</h1>
    );
  };
  
  export { GetAllOrdersForAdmin };
  