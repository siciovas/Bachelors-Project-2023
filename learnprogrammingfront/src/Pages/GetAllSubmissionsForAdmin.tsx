import {
  Accordion,
  AccordionButton,
  AccordionItem,
  AccordionPanel,
  Box,
  Flex,
  Spinner,
  useToast,
  Heading,
} from "@chakra-ui/react";
import { useCallback, useEffect, useState } from "react";
import { SubmissionTypes } from "../Types/SubmissionTypes";
import eventBus from "../Helpers/EventBus";
import { Unauthorized } from "../Constants/Auth";
import DOMPurify from "dompurify";
import { FaChevronDown } from "react-icons/fa";

const GetAllSubmissionsForAdmin = () => {
  const token = localStorage.getItem("accessToken");
  const [submissions, setSubmissions] = useState<SubmissionTypes[]>([]);
  const [isLoading, setIsLoading] = useState(true);
  const toast = useToast();

  const [expandedIndex, setExpandedIndex] = useState<number | null>(null);

  const handleAccordionClick = (index: number) => {
    if (index === expandedIndex) {
      setExpandedIndex(null);
    } else {
      setExpandedIndex(index);
    }
  };

  const getSubmissions = useCallback(async () => {
    const response = await fetch(`https://localhost:7266/api/submission`, {
      headers: {
        "Content-Type": "application/json",
        Authorization: `Bearer ${token}`,
      },
      method: "GET",
    });
    if (response.status === 401) {
      eventBus.dispatch("logOut", Unauthorized);
    } else if (response.status === 200) {
      const submission = await response.json();
      const reversedSubmission = [...submission].reverse();
      setSubmissions(reversedSubmission);
      setIsLoading(false);
    } else {
      toast({
        title: "Netikėta klaida",
        status: "error",
        duration: 5000,
        position: "top-right",
        isClosable: true,
      });
    }
  }, []);

  useEffect(() => {
    getSubmissions();
  }, [isLoading]);

  if (isLoading) {
    return (
      <Flex justifyContent="center" top="50%" left="50%" position="fixed">
        <Spinner size="xl" />
      </Flex>
    );
  }

  return (
    <>
      <Flex justifyContent={"center"}>
        <Heading mt={15} size="lg">
          Vartotojų prašymai
        </Heading>
      </Flex>
      <Accordion
        allowToggle
        width={"50%"}
        m={"auto"}
        mt={5}
        border={"transparent"}
        mb={7}
      >
        {submissions.map((submission, index) => (
          <AccordionItem>
            <AccordionButton
              onClick={() => handleAccordionClick(index)}
              _focus={{ boxShadow: "none" }}
              _hover={{ backgroundColor: "transparent" }}
              _active={{ backgroundColor: "transparent" }}
              backgroundColor={expandedIndex === index ? "gray.200" : "white"}
              borderRadius="full"
              border="1px"
              borderColor="gray.400"
              display="flex"
              justifyContent="flex-start"
              alignItems="center"
              px={4}
              py={2}
              transition="background-color 0.2s ease-out"
              _expanded={{ backgroundColor: "gray.200" }}
              _disabled={{ opacity: 0.4, cursor: "not-allowed" }}
              mt={2}
            >
              <Box
                as={FaChevronDown}
                mr={3}
                w={4}
                h={4}
                borderRadius="full"
                backgroundColor="gray.400"
              />
              {submission.name} {submission.surname} : {submission.topic}
            </AccordionButton>
            <AccordionPanel
              display={expandedIndex === index ? "block" : "none"}
              background={"white"}
              borderRadius={"50px 50px 50px 50px"}
            >
              <Box
                textAlign={"center"}
                dangerouslySetInnerHTML={{
                  __html: DOMPurify.sanitize(
                    submission.message as unknown as Node
                  ),
                }}
              ></Box>
            </AccordionPanel>
          </AccordionItem>
        ))}
      </Accordion>
    </>
  );
};

export { GetAllSubmissionsForAdmin };
