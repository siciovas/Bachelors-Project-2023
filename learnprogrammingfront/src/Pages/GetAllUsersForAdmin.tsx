import React from "react";
  
  const GetAllUsersForAdmin = () => {
  
    return (
        <h1>adminas visus userius matys cia</h1>
    );
  };
  
  export { GetAllUsersForAdmin };
  