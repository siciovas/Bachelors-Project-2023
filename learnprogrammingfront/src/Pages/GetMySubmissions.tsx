import {
  Accordion,
  AccordionButton,
  AccordionItem,
  AccordionPanel,
  Box,
  Flex,
  Spinner,
  Heading,
} from "@chakra-ui/react";
import { useCallback, useEffect, useState } from "react";
import { SubmissionTypes } from "../Types/SubmissionTypes";
import eventBus from "../Helpers/EventBus";
import { Unauthorized } from "../Constants/Auth";
import DOMPurify from "dompurify";
import toast from "react-hot-toast";
import { FaChevronDown } from "react-icons/fa";

const GetMySubmissions = () => {
  const token = localStorage.getItem("accessToken");
  const [submissions, setSubmissions] = useState<SubmissionTypes[]>([]);
  const [isLoading, setIsLoading] = useState(true);
  const [expandedIndex, setExpandedIndex] = useState<number | null>(null);

  const handleAccordionClick = (index: number) => {
    if (index === expandedIndex) {
      setExpandedIndex(null);
    } else {
      setExpandedIndex(index);
    }
  };
  const getSubmissions = useCallback(async () => {
    const response = await fetch(`https://localhost:7266/api/submission`, {
      headers: {
        "Content-Type": "application/json",
        Authorization: `Bearer ${token}`,
      },
      method: "GET",
    });
    if (response.status === 401) {
      eventBus.dispatch("logOut", Unauthorized);
    } else if (response.status === 200) {
      const submission = await response.json();
      setSubmissions(submission);
      setIsLoading(false);
    } else {
      toast.error("Netikėta klaida!");
    }
  }, []);

  useEffect(() => {
    getSubmissions();
  }, [isLoading]);

  if (isLoading) {
    return (
      <Flex justifyContent="center" top="50%" left="50%" position="fixed">
        <Spinner size="xl" />
      </Flex>
    );
  }

  return (
    <>
      <Flex justifyContent={"center"}>
        <Heading mt={15} size="lg">
          Mano prašymai
        </Heading>
      </Flex>
      <Accordion
        allowToggle
        width={"50%"}
        m={"auto"}
        mt={5}
        border={"transparent"}
        mb={7}
      >
        {submissions.map((submission, index) => (
          <AccordionItem>
            <AccordionButton
              onClick={() => handleAccordionClick(index)}
              _focus={{ boxShadow: "none" }}
              _hover={{ backgroundColor: "transparent" }}
              _active={{ backgroundColor: "transparent" }}
              backgroundColor={expandedIndex === index ? "gray.200" : "white"}
              borderRadius="full"
              border="1px"
              borderColor="gray.400"
              display="flex"
              justifyContent="flex-start"
              alignItems="center"
              px={4}
              py={2}
              transition="background-color 0.2s ease-out"
              _expanded={{ backgroundColor: "gray.200" }}
              _disabled={{ opacity: 0.4, cursor: "not-allowed" }}
              mt={2}
            >
              <Box
                as={FaChevronDown}
                mr={3}
                w={4}
                h={4}
                borderRadius="full"
                backgroundColor="gray.400"
              />
              {submission.topic}
            </AccordionButton>
            <AccordionPanel
              background={"white"}
              borderRadius={"50px 50px 50px 50px"}
              display={expandedIndex === index ? "block" : "none"}
            >
              <Box
                textAlign={"center"}
                dangerouslySetInnerHTML={{
                  __html: DOMPurify.sanitize(
                    submission.message as unknown as Node
                  ),
                }}
              ></Box>
            </AccordionPanel>
          </AccordionItem>
        ))}
      </Accordion>
    </>
  );
};

export { GetMySubmissions };
