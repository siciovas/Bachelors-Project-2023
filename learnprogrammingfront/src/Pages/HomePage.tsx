import React from "react";
import {
  Box,
  Stack,
  Heading,
  Text,
  Container,
} from "@chakra-ui/react";
import Slider from "react-slick";
import firstPhoto from "./WebPhotos/WebPhoto1.png";
import secondPhoto from "./WebPhotos/WebPhoto2.png";
import thirdPhoto from "./WebPhotos/WebElShopPhoto.png";

// Settings for the slider
const settings = {
  dots: false,
  arrows: false,
  fade: true,
  infinite: true,
  autoplay: true,
  autoplaySpeed: 5000,
  slidesToShow: 1,
  slidesToScroll: 1,
};

const HomePage = () => {
  const cards = [
    {
      title: "Neformalus ugdymas",
      text: "Pradėti mokytis programavimo niekada nėra per vėlu",
      image: firstPhoto,
    },
    {
      title: "Pasiruošimas egzaminui",
      text: "Programavimo uždaviniai skirti pasiruošti valstybiniui informacinių technologijų brandos egzaminui (VBE)",
      image: secondPhoto,
    },
    {
      title: "El. Parduotuvė",
      text: "Elektroninėje parduotuvėje rasite visas knygas, kurios padės įsisavinti programavimo žinias",
      image: thirdPhoto,
    },
  ];

  return (
    <Box
      className="sliderBox"
      height={"100vh"}
      overflow={"hidden"}
      pointerEvents="none"
    >
      {/* Slider */}
      <Slider {...settings}>
        {cards.map((card, index) => (
          <Box
            key={index}
            height={"6xl"}
            position="relative"
            backgroundPosition="center"
            backgroundRepeat="no-repeat"
            backgroundSize="cover"
            backgroundImage={`url(${card.image})`}
          >
            {/* This is the block you need to change, to customize the caption */}
            <Container height="600px" position="relative">
              <Stack
                spacing={6}
                w={"full"}
                maxW={"lg"}
                position="absolute"
                top="50%"
                transform="translate(0, -50%)"
              >
                <Heading
                  textColor={"white"}
                  fontSize={{ base: "3xl", md: "4xl", lg: "5xl" }}
                >
                  {card.title}
                </Heading>
                <Text textColor={"white"} fontSize={{ base: "md", lg: "lg" }} width={{base: "200px", md: "500px", }}>
                  {card.text}
                </Text>
              </Stack>
            </Container>
          </Box>
        ))}
      </Slider>
    </Box>
  );
};
export { HomePage };
