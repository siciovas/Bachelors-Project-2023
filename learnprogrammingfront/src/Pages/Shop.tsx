import React, { FormEvent, useCallback, useEffect, useState } from "react";
import {
  Box,
  Image,
  Badge,
  Grid,
  Flex,
  Spinner,
  Modal,
  ModalOverlay,
  ModalContent,
  ModalHeader,
  ModalFooter,
  ModalBody,
  ModalCloseButton,
  useDisclosure,
  Text,
  Button,
  Heading,
} from "@chakra-ui/react";
import { ShopTypes } from "../Types/ShopTypes";
import { useNavigate } from "react-router-dom";
import { AddNewShopItem } from "../Components/AddNewShopItem";
import { DeleteIcon } from "@chakra-ui/icons";
import { UserRole } from "../Constants/RolesConstants";
import toast from "react-hot-toast";
// @ts-ignore
import Fade from "react-reveal/Fade";

const Shop = () => {
  const navigate = useNavigate();
  const token = localStorage.getItem("accessToken");
  const [items, setItems] = useState<ShopTypes[]>([]);
  const [isLoading, setIsLoading] = useState(true);
  const role = localStorage.getItem("role");
  const { isOpen, onOpen, onClose } = useDisclosure();
  const [deletingId, setDeletingId] = useState<number>();

  const NavigateToItem = (shopItemId: number) => {
    navigate("/preke", {
      state: {
        shopItemId: shopItemId,
      },
    });
  };

  const openModal = (id: number) => {
    setDeletingId(id);
    onOpen();
  };

  const AddShopItem = useCallback(
    async (
      e: FormEvent<HTMLFormElement>,
      photo: string,
      name: string,
      price: number,
      description: string,
      pageNumber: number,
      language: string,
      bookCoverType: string,
      publisher: string,
      releaseDate: string
    ): Promise<void> => {
      e.preventDefault();
      const response = await fetch("https://localhost:7266/api/shop", {
        headers: {
          "Content-Type": "application/json",
          Authorization: `Bearer ${token}`,
        },
        method: "POST",
        body: JSON.stringify({
          photo,
          name,
          price,
          description,
          pageNumber,
          language,
          bookCoverType,
          publisher,
          releaseDate,
        }),
      });
      if (response.status === 201) {
        setIsLoading(true);
        toast.success("Prekė pridėta!");
        onClose();
      } else {
        toast.error("Nepavyko!");
      }
    },
    []
  );

  const getShopItems = useCallback(async () => {
    const response = await fetch(`https://localhost:7266/api/shop`, {
      headers: {
        "Content-Type": "application/json",
        Authorization: `Bearer ${token}`,
      },
      method: "GET",
    });
    const allItems = await response.json();
    const reversedItems = [...allItems].reverse();
    setItems(reversedItems);
    setIsLoading(false);
  }, []);

  useEffect(() => {
    getShopItems();
  }, [isLoading]);

  const deleteShopItem = async (
    e: React.MouseEvent<HTMLButtonElement, MouseEvent>,
    shopItemId: number
  ): Promise<void> => {
    e.preventDefault();
    const response = await fetch(
      `https://localhost:7266/api/shop/${shopItemId}`,
      {
        headers: {
          "Content-Type": "application/json",
          Authorization: `Bearer ${token}`,
        },
        method: "DELETE",
      }
    );

    if (response.status === 204) {
      setIsLoading(true);
      toast.success("Prekė ištrinta!");
      onClose();
    } else {
      toast.error("Nepavyko ištrinti!");
    }
  };

  if (isLoading) {
    return (
      <Flex justifyContent="center" top="50%" left="50%" position="fixed">
        <Spinner size="xl" />
      </Flex>
    );
  }

  return (
    <>
      <Flex mt={5} justify="center">
        <Heading size="lg" fontFamily={"Roboto"}>El. Parduotuvė</Heading>
      </Flex>
      {role === UserRole.Admin && <AddNewShopItem AddShopItem={AddShopItem} />}
      <Grid
        margin={20}
        templateColumns={{
          base: "repeat(1, 1fr)",
          md: "repeat(2, 1fr)",
          lg: "repeat(3, 1fr)",
          xl: "repeat(3, 1fr)",
          "2xl": "repeat(4, 1fr)",
        }}
        justifyContent={"center"}
        gap={3}
        mt={10}
      >
        {items.map((item) => {
          return (
            <Fade left>
              <Box
                borderWidth="1px"
                borderRadius="lg"
                overflow="hidden"
                borderColor={"black"}
                width={"325px"}
                bg={"whitesmoke"}
              >
                <Image
                  cursor={"pointer"}
                  onClick={() => NavigateToItem(item.id)}
                  height="400px"
                  width="100%"
                  src={"data:image/jpeg;base64," + item.photo}
                />

                <Box p="6">
                  <Box display="flex" alignItems="baseline">
                    <Badge borderRadius="full" px="2" colorScheme="teal">
                      {item.price} €
                    </Badge>
                    <Box
                      color="gray.500"
                      fontWeight="semibold"
                      letterSpacing="wide"
                      fontSize="xs"
                      textTransform="uppercase"
                      ml="2"
                    >
                      {item.pageNumber} puslapių &bull; {item.language} kalba
                    </Box>
                  </Box>

                  <Box
                    mt="1"
                    fontWeight="semibold"
                    as="h4"
                    lineHeight="tight"
                    noOfLines={1}
                    textTransform={"uppercase"}
                  >
                    {item.name}
                  </Box>
                  {role === UserRole.Admin && (
                    <>
                      <Flex justify="flex-end">
                        <DeleteIcon
                          cursor={"pointer"}
                          color={"red.500"}
                          onClick={() => openModal(item.id)}
                        />
                      </Flex>
                    </>
                  )}
                </Box>
              </Box>
            </Fade>
          );
        })}
      </Grid>
      <>
        <Modal isOpen={isOpen} onClose={onClose}>
          <ModalOverlay />
          <ModalContent>
            <ModalHeader>Perspėjimas!</ModalHeader>
            <ModalCloseButton />
            <ModalBody>
              <Text>Ar tikrai norite ištrinti prekę?</Text>
            </ModalBody>

            <ModalFooter>
              <Button
                background="red.500"
                mr={3}
                onClick={(e) => deleteShopItem(e, deletingId as number)}
                color="white"
                _hover={{
                  bg: "red.700",
                }}
              >
                Ištrinti
              </Button>
            </ModalFooter>
          </ModalContent>
        </Modal>
      </>
    </>
  );
};

export { Shop };
