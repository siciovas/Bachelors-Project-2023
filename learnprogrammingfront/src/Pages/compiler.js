/* eslint-disable no-undef */
export const outf = (text) => {
  console.log("Result:", text);
  var mypre = document.getElementById("output");
  mypre.value = mypre.innerHTML + text;
};

export const run = (code) => {
  console.log(code);
  Sk.pre = "output";
  Sk.configure({ output: outf });
  var compiledCode = Sk.importMainWithBody("<stdin>", false, code);
  Sk.misceval.asyncToPromise(function() {
    return Sk.misceval.callsim(compiledCode.$d.run, Sk.ffi.remapToPy([5, 5, 15, 15]));
  }).then(function(result) {
    console.log("labas")
    console.log(result);
    // compare output with expected output
    if (output === expectedOutput) {
      numPassed++;
    }
  }, function(err) {
    console.log(err.toString());
  });
};
