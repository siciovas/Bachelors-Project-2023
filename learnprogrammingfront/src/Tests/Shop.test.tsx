import React from "react";
import { mount } from "enzyme";
import { Shop } from "../Pages/Shop";
import { BrowserRouter } from "react-router-dom";

describe("Shop", () => {

  it("renders without errors", () => {
    const wrapper = mount(<BrowserRouter><Shop /> </BrowserRouter>);
    expect(wrapper).toMatchSnapshot();
  });
});
