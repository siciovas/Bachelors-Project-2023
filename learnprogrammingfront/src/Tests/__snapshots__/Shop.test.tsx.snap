// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shop renders without errors 1`] = `ReactWrapper {}`;
